packages/valory/skills/abstract_round_abci/tests/test_base.py

```diff
@@ -540,20 +550,75 @@ def test_init(self, data: Dict, setup_data: Optional[Dict]) -> None:
                     f"AbciAppDB data must be `Dict[str, List[Any]]`, found `{type(data)}` instead"
                 ),
             ):
-                AbciAppDB(setup_data=data)
-        else:
-            db = AbciAppDB(setup_data=data)
-            assert db._data == {0: setup_data}
-            assert db.setup_data == setup_data
-            assert db.cross_period_persisted_keys == []
+                AbciAppDB(data, cross_period_persisted_keys)
+            return
+
+        # use copies because otherwise the arguments will be modified and the next test runs will be polluted
+        data_copy = deepcopy(data)
```

Comment:
Not immediately obvious why the tests had to change

Reply:
The tests were updated in order to reveal the mutability issue. No line was removed, only added extra checks.

The `deepcopy` in the line you commented on seems strange at first, but it is only there in order to prevent modifying the arguments given with `pytest.mark.parametrize` among tests. Notice how the `data_copy` are only used for the db initialization and also for trying to indirectly modify it.

I did not have the chance to add comments to the PR yet, otherwise would have highlighted this.
73d18ce0d88beefb616a885ee636ae538c52f0ad introduced these tests, which reveal the db issue in [Fix mutability issue with synchronized db #5893](https://github.com/valory-xyz/open-autonomy/actions/runs/3429498122/jobs/5715425857).
Then b267af9892f007cdecdff4066cfda789d621bae0 fixes the database.